main.cpp

```diff
@@ -56,15 +56,21 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int val, const char* str): value{val}, name{str} 
+    {
+
+    } //1
+    int value;
+    std::string name;
 };
 
-struct <#structName1#>                                //4
+struct CompareDaFunk                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
+        if (a == nullptr || b == nullptr)
+            return nullptr;
+            
         if( a->value < b->value ) return a;
         if( a->value > b->value ) return b;
         return nullptr;
```

Comment:
You now have 2 `return nullptr;` statements.
`Don't Repeat Yourself`
revise your logic so that you only have one `return nullptr` statement.
- [ ] revise

Reply:
I generally avoid doing it this way because if that CompareDaFunc function body was long, it results in an indent of the whole section, sometimes spilling over the right side in the editor.  If I have more "if this is true, then execute the following", then that results in more indented code.  Instead, if the check fails, I try to get out of dodge, which results in not having to indent the subsequent code in curly braces.  But that's my preference and my obligation here is to comply with the project manager's preferences.